lib/pipefy_message/providers/aws_client/sns_broker.rb

```diff
@@ -27,18 +28,28 @@ def initialize(_opts = {})
         ##
         # Publishes a message with the given payload to the SNS topic
         # with topic_name.
-        def publish(payload, topic_name)
+        def publish(payload, topic_name, context = "NO_CONTEXT_PROVIDED")
           message = prepare_payload(payload)
           topic_arn = @topic_arn_prefix + (@is_staging ? "#{topic_name}-staging" : topic_name)
           topic = @sns.topic(topic_arn)
 
           logger.info(
             { topic_arn: topic_arn,
               payload: payload,
-              message_text: "Attempting to publish a json message to topic #{topic_arn}" }
+              message_text: "Attempting to publish a json message to topic #{topic_arn}}" }
           )
 
-          result = topic.publish({ message: message.to_json, message_structure: " json " })
+          result = topic.publish({ message: message.to_json, message_structure: " json ",
+                                   message_attributes: {
+                                     "correlationId" => {
```

Comment:
Is this `correlationId` really a correlation ID or is going to be refactored in the future?
As for the current implementation, it seems that's just a random value not correlated with anything else.

Reply:
IMHO it's a really correlation ID but for now, we are just generating it, later we'll use that on the message retries headers and logs tracing.

But the thing is... we have to implement that to use later, if you think that we don't need that please share your thoughts.